### PR TITLE
feat: always scan the root package of current module/project for endpoints

### LINKED ISSUE #994: Consider always scaning current project/module root package for endpoints
Currently, the packages from current project (or current module of a multi-module project) is scanned for endpoints, only if there is no configuration for the exposed packages from dependencies (or other modules). If the user configure the maven/gradle plugin to scan and include a single external endpoint from some dependency (e.g. SSO Kit), then they should also always list the a high-level package from the current project/module which is not convenient and redundant IMO.

Configuring the for exposed packages from other modules or from dependencies should not affect the parser behavior for scanning the packages in the current project (or current module of a multi-module project), as most of the times, the use case is to use some external utility endpoints, and not to hide the endpoints of the current project/module. 

We should consider always scanning the current project/module's packages, plus the additional exposed packages from the dependencies or other modules.

### PR DESCRIPTION
**Pull request by taefi**
## Description

This is a fixed-term solution for always including the root package of current 
module/project while scanning for endpoints. 

Finding the root package of the current module can be done in different ways,
and in this PR it is relying on the existence of a class annotated with 
@SpringBootApplication.

The better approach for scanning the endpoints could rely on spring beans instead, 
however, at the moment making such a breaking change is not possible, so until
then, this can be considered as a partial solution.

Fixes #994 

## Type of change

- [ ] Bugfix
- [x] Feature

## Checklist

- [x] I have read the contribution guide: https://vaadin.com/docs/latest/guide/contributing/overview/
- [x] I have added a description following the guideline.
- [x] The issue is created in the corresponding repository and I have referenced it.
- [x] I have added tests to ensure my change is effective and works as intended.
- [x] New and existing tests are passing locally with my change.
- [x] I have performed self-review and corrected misspellings.

#### Additional for `Feature` type of change

- [x] Enhancement / new feature was discussed in a corresponding GitHub issue and Acceptance Criteria were created.

### CODE REVIEW

File: packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java [hunk 1]
```diff
@@ -1,18 +1,12 @@
...
 
 import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.HashSet;
-import java.util.List;
-import java.util.Objects;
-import java.util.Set;
+import java.util.*;
```
Comment (Artur-) [inline packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java:4]: Expand imports, maybe we could have a check for this...

File: packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java [hunk 2]
```diff
@@ -285,24 +286,29 @@ public OpenAPI execute() {
...
         var storage = new SharedStorage(config);
 
-        var classGraph = new ClassGraph().enableAnnotationInfo()
-                .ignoreClassVisibility()
-                .overrideClassLoaders(config.getClassLoader());
+        var buildDirectories = config.getClassPathElements().stream()
+                .filter(e -> !e.endsWith(".jar")).toList();
 
-        Collection<String> packages = config.exposedPackages;
+        Collection<String> packages = getConfig().getExposedPackages();
```
Comment (Artur-) [inline packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java:292]:
```suggestion
        Collection<String> packages = config.getExposedPackages();
```

File: packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java [hunk 3]
```diff
@@ -482,4 +488,27 @@ public Collection<Plugin> getPlugins() {
...
+        public Optional<String> findRootPackage(List<String> buildDirectories) {
+
+            var currentModuleClassGraph = new ClassGraph()
+                    .enableAnnotationInfo().ignoreClassVisibility()
+                    .overrideClassLoaders(ClassLoader.getSystemClassLoader())
+                    .overrideClasspath(buildDirectories);
+
+            String springBootApplicationAnnotation = "org.springframework.boot.autoconfigure.SpringBootApplication";
+
+            try (var scanResult = currentModuleClassGraph.scan()) {
```
Comment (Artur-) [inline packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java:503]: Does scanning everything again have some performance impact?

**Review by Artur- — CHANGES_REQUESTED** [review]

File: packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java (hunk 3 above)
Comment (taefi) [reply packages/java/parser-jvm-core/src/main/java/dev/hilla/parser/core/Parser.java:503]: This is only looking into the `classes` directory and is not scanning the dependencies.

**Pull request closed without merging**